app/models/system-state.server.ts

```diff
@@ -0,0 +1,23 @@
+import { StateEnumType } from '@prisma/client';
+
+import { prisma } from '~/db.server';
+
+import type { SystemState } from '@prisma/client';
+
```

Comment:
I think this file could use a comment.  Something like:

```ts
/**
 * The SystemState represents global (i.e., cross-instance), shared settings among Starchart instance(s).
 * Initially, this includes info on whether or not the DNS Reconciler needs to be run at the next opportunity.
 * We use this global SystemState to make sure that we never have more than 1 instance of the reconciler
 * running at a time.
 */
```

Fix that to be be what makes sense to you, but let's try to explain it to future readers.

Reply:
Actually cross-instance concurrency is inhibited in the worker, when I'm checking the number of running workers in the queue. This one is only to not keep polling Route53 every 2 minutes if we know nothing has changed

```
 * We use this global SystemState to make sure that we stop hitting the Route53 API when no changes were made
 * to our Records table`
```